packages/primitives/src/primitives/fetch.js

```diff
@@ -125,9 +125,17 @@ function addDuplexToInit(init) {
  * Export fetch with an implementation that uses a default global dispatcher.
  * It also re-cretates a new Response object in order to allow mutations on
  * the Response headers.
+ * @param {RequestInfo} info
+ * @param {RequestInit} init
  */
 export async function fetch(info, init) {
   init = addDuplexToInit(init)
+  if (info instanceof Request) {
+    init.method ??= info.method
```

Comment:
I am not sure how far long this PR is, so the comment may be premature, but two thoughts:

1. why are only these particular properties of `Request` are copied over. It has other many other properties which are also valid to be set via the second argument to `fetch`. E.g. `mode`, `cache`, and others. 
2. I am not sure how this fixes https://github.com/vercel/next.js/issues/57905. That issue is about the `Request` constructor not working properly when the first argument to it is a `Request` instance - `fetch` is not involved at all, though it is in some ways similar.